build-logic/src/main/kotlin/reflections-manifest.gradle.kts

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import java.net.URLClassLoader
+
+tasks.register("cacheReflections") {
+    description = "Caches reflection output to make regular startup faster. May go stale and need cleanup at times."
+
+    val sourceSets = project.convention.getPlugin(JavaPluginConvention::class.java).sourceSets
+    val mainSourceSet: SourceSet = sourceSets[SourceSet.MAIN_SOURCE_SET_NAME]
+
+    inputs.files(mainSourceSet.output.classesDirs)
+    dependsOn(tasks.named("classes"))
+    outputs.upToDateWhen { tasks.named("classes").get().state.upToDate }
+
+    // TODO: output to the task's own build directory, because putting things in the
+    //     classes directory confuses gradle caching.
+    val manifestFile = File(mainSourceSet.output.classesDirs.first(), "manifest.json")
+    outputs.file(manifestFile)
+
+    doLast {
+        val classPaths = mainSourceSet.compileClasspath.map { it.toURI().toURL() }
+        val classLoader = URLClassLoader(classPaths.toTypedArray())
+//        try {
+//            val reflections = Reflections(ConfigurationBuilder()
+//                .setSerializer(JsonSerializer())
+//                .addClassLoader(classLoader)
+//                // .filterInputsBy(FilterBuilder.parsePackages("+org"))
+//                .addUrls(inputs.files.map { it.toURI().toURL() })
+//                .setScanners(TypeAnnotationsScanner(), SubTypesScanner(false)))
+//            reflections.save(manifestFile.toString())
```

Comment:
FIXME: decide if we're re-enabling cacheReflections task or not

Reply:
doesn't gestalt depend on reflections? how is it working without reflections.

Reply:
It does depend on Reflections, but if there is not a pre-compiled Reflections cache it will do the scan at runtime: https://github.com/MovingBlocks/gestalt/blob/a51ba072f2e7c8e084ddf0b530d73bbb713ae2a3/gestalt-module/src/main/java/org/terasology/gestalt/module/ModuleFactory.java#L211-L225

Reply:
the TypeRegistry in nui.reflections already relies exclusively on the runtime scan: https://github.com/MovingBlocks/TeraNUI/blob/0d6d07dcfbd29f4319eb6084de95d24c81ba53a9/nui-reflect/src/main/java/org/terasology/reflection/TypeRegistry.java#L91-L93